lib/WUI/nhttp/req_parser.cpp

```diff
@@ -210,28 +212,31 @@ bool RequestParser::uri_filename(char *buffer, size_t buffer_size) const {
         len = amp - url.begin();
     }
 
-    // Make sure the user is not able to "escape" from directory.
+    if (buffer_size < len) {
+        return false;
+    }
```

Comment:
:thinking: I wonder… it's probably not worth fixing/supporting right now, but maybe a note would not go amiss. If we have an URL with a bunch of `%??` in it, the result is shorter. So this is a bit stricter than necessary.